src/content/ui/interactivity/focus.md

```diff
@@ -3,179 +3,222 @@ title: Understanding Flutter's keyboard focus system
-  around the widget subtree you would like to listen to, and
-  set the `onKeyEvent` attribute of the widget to your handler.
-  Set `canRequestFocus: false` on the widget if
-  you also don't want it to be able to take primary focus.
-  This is because the `onKeyEvent` attribute on the `Focus` widget can be
-  set to something else in a subsequent build, and if that happens,
-  it overwrites the `onKeyEvent` handler you set on the node.
-- Do call `requestFocus()` on a node to request that it receives the
-  primary focus, especially from an ancestor that has passed a node it owns to
-  a descendant where you want to focus.
-- Do use `focusNode.requestFocus()`. It is not necessary to call
-  `FocusScope.of(context).requestFocus(focusNode)`. The
-  `focusNode.requestFocus()` method is equivalent and more performant.
-
-### Unfocusing
-
-There is an API for telling a node to "give up the focus", named
-`FocusNode.unfocus()`. While it does remove focus from the node, it is important
-to realize that there really is no such thing as "unfocusing" all nodes. If a
-node is unfocused, then it must pass the focus somewhere else, since there is
-_always_ a primary focus. The node that receives the focus when a node calls
-`unfocus()` is either the nearest `FocusScopeNode`, or a previously focused node
-in that scope, depending upon the `disposition` argument given to `unfocus()`.
-If you would like more control over where the focus goes when you remove it from
-a node, explicitly focus another node instead of calling `unfocus()`, or use the
-focus traversal mechanism to find another node with the `focusInDirection`,
+**Do create `FocusNode` and `FocusScopeNode` objects in a stateful widget.**
+: When you finish using `FocusNode` and `FocusScopeNode`,
+  you need to dispose of them.
+  For this reason, only create them inside of a stateful widget's `State` object.
+  You can override the stateful widget's `dispose` method to dispose of the
+  focus objects.
+
+**Do set the `debugLabel` of a focus node widget.**
+: This helps with diagnosing focus issues.
+
+**Do call the `requestFocus()` method on a node to request the primary focus.**
+: Call this method from an node that has passed a node
+  it owns to a descendant where you want to the focus.
+
+**Do use `focusNode.requestFocus()`.**
+: You don't need to call `FocusScope.of(context).requestFocus(focusNode)`.
+  The `focusNode.requestFocus()` method works the same with better performance.
+
+**Don't allocate a new `FocusNode` object for each build.**
+: This can cause memory leaks.
+  It might also cause a loss of focus when the widget rebuilds while
+  the node has focus.
+
+**Don't use the same `FocusNode` for multiple widgets.**
+: If you do, the widgets fight over managing the attributes of the node.
+  That might result in an unexpected outcome.
+
+**Don't set the `onKeyEvent` callback on a `FocusNode` that `Focus` manages.**
+: To set focus to respond to a key event, use an `onKeyEvent` handler.
```

Comment:
This paragraph is about how to set up an onKeyEvent handler, but the heading is a warning about setting it on the node. They don't quite go together. I think maybe keep this paragraph, but change the heading, since that's covered by the `withExternalFocusNode` discussion?

Reply:
See previous.